### PR TITLE
data: remove /bin/sh from snapd.sh

### PR DESCRIPTION
**Pull request by zyga**
The snapd.sh file is meant to be sourced so the shebang line is just
some meta-data that shellcheck is using. This meta-data is, however,
also picked up by rpmlint, resulting in the following warning:

[   73s] snapd.x86_64: W: sourced-script-with-shebang /etc/profile.d/snapd.sh /bin/sh --this-shebang-is-just-here-to-inform-shellcheck--
[   73s] This text file contains a shebang, but is meant to be sourced, not executed.

This patch keeps the shebang for validation but drops it for
installation.

Signed-off-by: Zygmunt Krynicki <zygmunt.krynicki@canonical.com>

### CODE REVIEW

**Comment by ogra1** [comment]
note that you can run shellcheck with the -s parameter for files without shebang like: "shellcheck -s sh /path/to/file" or "shellcheck -s bash /path/to/file" (whatever shell you want it to check)

File: data/env/Makefile [hunk 1]
```diff
@@ -20,6 +20,14 @@ ENVD := /etc/profile.d
 	sed   < $< > $@ \
 		s:@SNAP_MOUNT_DIR@:${SNAP_MOUNT_DIR}:g
 
+# When processing snapd.sh.in remove the /bin/sh shebang as this file is meant
+# to be sourced, not executed, and the presence of that line causes rpmlint
+# validation warnings.
+snapd.sh: snapd.sh.in
```
Comment (mvo5) [inline data/env/Makefile]: Can we remove it in snapd.sh directly?

Comment (zyga) [reply data/env/Makefile]: I think we can although shellcheck might complain about that. We could use ogra's advice as well but I don't know how easy that is.

Comment (zyga) [reply data/env/Makefile]: Without the shebang entirely:

```
zyga@tumbleweed:~/snapd/data/env> shellcheck snapd.sh

In snapd.sh line 1:
# Expand $PATH to include the directory where snappy applications go.
^-- SC2148: Tips depend on target shell and yours is unknown. Add a shebang.


In snapd.sh line 3:
if [ -n "${PATH##*${snap_bin_path}}" -a -n "${PATH##*${snap_bin_path}:*}" ]; then
                                     ^-- SC2166: Prefer [ p ] && [ q ] as [ p -a q ] is not well defined.


In snapd.sh line 17:
if [ -n "${XDG_DATA_DIRS##*${snap_xdg_path}}" -a -n "${XDG_DATA_DIRS##*${snap_xdg_path}:*}" ]; then
                                              ^-- SC2166: Prefer [ p ] && [ q ] as [ p -a q ] is not well defined.

```

Without the shebang but with explicit file type (not sure how to pass this elsewhere)

```
zyga@tumbleweed:~/snapd/data/env> shellcheck -s sh snapd.sh

In snapd.sh line 3:
if [ -n "${PATH##*${snap_bin_path}}" -a -n "${PATH##*${snap_bin_path}:*}" ]; then
                                     ^-- SC2166: Prefer [ p ] && [ q ] as [ p -a q ] is not well defined.


In snapd.sh line 17:
if [ -n "${XDG_DATA_DIRS##*${snap_xdg_path}}" -a -n "${XDG_DATA_DIRS##*${snap_xdg_path}:*}" ]; then
                                              ^-- SC2166: Prefer [ p ] && [ q ] as [ p -a q ] is not well defined.
```

Sadly I think this is the best we can easily do. We should also address those two issues.

File: data/env/Makefile [hunk 2]
```diff
@@ -20,6 +20,14 @@ ENVD := /etc/profile.d
...
+	sed   < $< \
```
Comment (bboozzoo) [inline data/env/Makefile]: `sed -e s:@SNAP_MOUNT_DIR@:${SNAP_MOUNT_DIR}:g -e '1d;2d' ..` should do it in one go

File: data/env/Makefile (hunk 1 above)
Comment (bboozzoo) [reply data/env/Makefile]: Would adding this at the top of snapd.sh work?
```
# shellcheck shell=sh
```

**Pull request merged**